### DISCUSSION TITLE
Sumatra Not printing command line but not failing either

### DISCUSSION THREAD

**Discussion by otiiraphael** [sumatrapdfreader/sumatrapdf#4551] [General]
I have a weird issue, when prinitng from my desktop powershell i can print to the print server using this syntax
SumatraPDF-3.5.2-64.exe -print-to "\\printserver\printername" -print-settings noscale "pathtofile\filename.pdf"

but when i run the same thing on Powershell on our SQL server it does not error and the -console shows exit code 0 but nothing comes out of the printer. I'm using the portable version on both machines with files that both machines can access. 

i don't know where else i can see what might be causing the issue

**Comment by GitHubRulesOK** [comment]
SumatraPDF literally hands off images of pages to Windows without any significant feedback in the downstream process.
Basically it works as a console user heads-up (serverless) application. So have you tried Acrobat or other user based PDF application in that configuration.

For error message you need to watching the cmd console where a popup might appear..

There are several likely points of failure as PDF print needs the GDI/GUI/GPU system to render into the graphics processor and some server configurations dont expect / allow user based graphical applications. 

There is also the User Authorisation aspect as SumatraPDF need to be able to talk to itself via backchannels and those may not be open or unrestricted in a server.

Use Ghostscript for served print handling it only needs a minimal graphics processor interaction and has comprehensive error feedback.

**Comment by kjk** [comment]
Probably because you're running it from a service and that's not supported.

You can try to get / provide us with more detail:
* get latest [pre-release](https://www.sumatrapdfreader.org/prerelease) because I've added a lot of logging to printing process there
* add `-log` option. This will log to a file `sumatra-log.txt` in `%LOCALAPPDATA%\SumatraPDF-data` directory
* at exit it should open the log file in notepad but it might not work in your scenario in which case you need to find `sumatra-log.txt` file
* alternatively, you can download and run [logview](https://www.sumatrapdfreader.org/docs/Logview) before starting Sumatra. that way you can see SumatraPDF logs in logview.

If you post the logs here I'll have more data to maybe diagnose the issue.

**Reply by GitHubRulesOK** [reply]
Hmm the logview download link does not work for me?

**Reply by kjk** [reply]
Oops, thanks for letting me know, have to fix it.

**Comment by otiiraphael** [comment]
That is the log output 

PrintToDevice: finished ok
PrintFile2: finished ok
PrintFile: finished ok
Finished printing, exitCode: 0
Exiting with exit code: 0

For Prerel i cannot get a -log to output a file but -console shows the same as stable -log output 

My hope was to run a sql agent job to launch powershell script but i cannot even get it to work within PS or CMD on that machine even though i mine it works fine. there might be some network restrictions or other dependancies that im not aware that it needs. its running sql2014 so machine is a bit older. I will see if our newer sql server also has this issue

**Reply by kjk** [reply]
What it tells me is that from the point of view of SumatraPDF we did rasterize and sent the bitmap to the printer and the OS told us that it worked.

But it didn't. The reality is that printing is merely sending to a printer spool. The actual printing happens asynchronously.

I have no idea why it doesn't work. You can check printer spool, system event logs for some clues.

There were other reports that printing from inside a process that is running as a service doesn't work. It's not terribly surprising given there are restrictions in service. I assume your sqlserver does run as a service.

**Comment by otiiraphael** [comment]
Thanks for that Krzysztof, This does seem print driver-related (Event logs seem to hint at that though not explicitly) i will add the printer locally with the driver and see if that resolves it. 
I was just trying through powershell alone on the sql server before testing the sql service part so i know its not that. 

Thanks again.
